### ISSUE TITLE
Unexpected parse problems in valid JS code when arrow functions are used as values in object surrounded with parentheses

### ISSUE THREAD

**Issue by IllusionMH** [microsoft/TypeScript#32914]
**TypeScript Version:**  3.7.0-dev.20190815 (there are same errors in playground even in 2.x)

**Search Terms:** Identifier expected , arrow function in object

**Code**

`test.js`
```js
const value = 1;
const a = {
    b: () => false
}

const test = () => ({
    // "Identifier expected." error on "!" and two "Duplicate identifier '(Missing)'." errors on space.
    prop: !value, // remove ! to see that errors will be gone
    run: () => { //replace arrow function with regular function to see that errors will be gone
        // comment next line or remove "()" to see that errors will be gone
        if(!a.b()) { return 'special'; }

        return 'default';
    }
});
```
There are no look-alike Unicode characters.
There are no problems if you switch order of `prop` and `run`.

Error is reproducible even when I write
```js
const test = ({
    prop: !value
    run: () => {
        if(!a.b()) { return 'special'; }

        return 'default';
    },
});
```
But works as expected if I remove parentheses around object or replace arrow function with regular one.

**Expected behavior:** No errors (code is valid and produces expected results in browsers)

**Actual behavior:** A lot of errors, most notably `Identifier expected.` on `!` character in line `prop: !value`

**UPD.** Object properties are actually parsed as arrow function parameters:
![image](https://user-images.githubusercontent.com/1312662/63108038-a968f500-bf8e-11e9-9f21-ecedf9b9ab97.png)

**Playground Link:** [Example](https://www.typescriptlang.org/play/index.html?noImplicitAny=false&target=6#code/MYewdgzgLgBAbgQwDYFcCmMC8MCMBuAKFElgSxgG8CYaYAjALhgAoBKLAPhgDNkI0CAX0JFw0GFDTQc5NpxZVaMAPTKYAIgCSAEzRgoAS24G0AJxhoAHgAc0wSdoB06i6dMhz4DQEIXCMNoSAO4gGgAiKNZIBsAIkjAGuvpGJuYA5MwAsgYQEAZgAOasac6u7qYQMF4Q1gjAaI7UtNbu1kzeiKhoADQqaqZoALYgcBjeEqH8GFAAFnFlHpVBBkhI9BgF4AJKpihgTHKYXBR9A1F1GAhuIEE8e-YGXsuzMAMFKEhXd2APXlCTaGmc1gZnKSxWazoGy2TSUNFUMFAg0GelgYCssGi6Kq5gGw1GGjYLn+MCmEmBCwqMGWq3WME26NhcKMzG8CEcdDY7BOAygKFMYBgaRqdgMyDSeBgggITJ2aD5AqFul4HygEqZgm6QlYhCAA) (I disabled `noImplicitAny` to remove noise)

**Related Issues:** Not found

**Comment by DanielRosenwasser** [comment]
Jeez - this is likely some bad lookahead logic which makes the compiler think that it's parsing a destructured parameter.

**Comment by IllusionMH** [comment]
Yeah, it is a bit strange that it picks destructured parameter when it would have inappropriate `!`, but it's a `ParenthesizedExpression > ObjectLiteralExpression` as soon as `!` is removed.

**Comment by RyanCavanaugh** [comment]
Thanks for the good repro with details!

**Comment by RyanCavanaugh** [comment]
Additional testcase, very similar
```ts
const graphql = () => ({
  mutations: ``,
  follow: () => {
    if (isNaN(0)) {
    }
  }
});
```
